orchestrator/relayer/src/batch_relaying.rs

```diff
@@ -193,27 +243,47 @@ async fn submit_batches(
                 }
                 let cost = cost.unwrap();
                 info!(
-                "We have detected latest batch {} but latest on Ethereum is {} This batch is estimated to cost {} Gas / {:.4} ETH to submit",
-                latest_cosmos_batch_nonce,
-                latest_ethereum_batch,
-                cost.gas_price.clone(),
-                downcast_to_u128(cost.get_total()).unwrap() as f32
-                    / downcast_to_u128(one_eth()).unwrap() as f32
-            );
+                    "We have detected latest batch {} but latest on Ethereum is {} This batch is estimated to cost {} Gas / {:.4} ETH to submit",
+                    latest_cosmos_batch_nonce,
+                    latest_ethereum_batch,
+                    cost.gas_price.clone(),
+                    downcast_to_u128(cost.get_total()).unwrap() as f32
+                        / downcast_to_u128(one_eth()).unwrap() as f32
+                );
 
-                let res = send_eth_transaction_batch(
-                    current_valset.clone(),
-                    oldest_signed_batch,
-                    &oldest_signatures,
-                    web3,
-                    timeout,
-                    gravity_contract_address,
-                    gravity_id.clone(),
-                    ethereum_key,
-                )
-                .await;
-                if res.is_err() {
-                    info!("Batch submission failed with {:?}", res);
+                // TODO: Convert the other methods to this style
+                let should_relay = if config.batch_market_enabled {
+                    should_relay_batch(
+                        web3,
+                        &oldest_signed_batch,
+                        cost.get_total(),
+                        our_ethereum_address,
+                    )
+                    .await
+                } else {
+                    true
+                };
+
+                if should_relay {
+                    let res = send_eth_transaction_batch(
+                        current_valset.clone(),
+                        oldest_signed_batch,
+                        &oldest_signatures,
+                        web3,
+                        timeout,
+                        gravity_contract_address,
+                        gravity_id.clone(),
+                        ethereum_key,
+                    )
+                    .await;
+                    if res.is_err() {
+                        info!("Batch submission failed with {:?}", res);
+                    }
+                } else {
+                    info!(
+                        "Not relaying batch due to it not being profitable: {:?}",
+                        oldest_signed_batch
```

Comment:
👋 Might want to include cost and reward in this log statement to improve observability on _why_ it didn't relay.